lms/static/scripts/frontend_apps/services/client-rpc.ts

```diff
@@ -124,15 +132,12 @@ export class ClientRPC extends TinyEmitter {
 
     this._server.register(
       'reportActivity',
-      /**
-       * @param {AnnotationEventType} eventType
-       * @param {AnnotationEventData} data
-       */
-      (eventType, data) => {
+      (eventType: AnnotationEventType, data: AnnotationEventData) => {
         this.emit('annotationActivity', eventType, data);
       }
     );
 
+    this._resolveGroups = () => {};
```

Comment:
This is a dummy assignment to satisfy TS that `_resolveGroups` is initialized in the constructor. It doesn't know that `new Promise` invokes its callback synchronously and thus will execute assignments before the constructor returns.